File: backend/app/domain/schemas/application_type.py
from typing import List, Any, Optional

from pydantic import BaseModel


class StatusFlux(BaseModel):
    status: str
    scope: list[int]


class ApplicationTypeBase(BaseModel):
    name: str
    description: str
    status_flux: list[StatusFlux]
    who_can: list[int]


class ApplicationTypeCreate(ApplicationTypeBase):
    pass


class ApplicationTypeUpdate(ApplicationTypeBase):
    pass


class ApplicationTypeInDB(ApplicationTypeBase):
    id: int

    class Config:
        orm_mode = True


class ApplicationTypeResponse(ApplicationTypeInDB):
    application_sub_type: Optional[List[Any]]

File: client/src/app/core/interfaces/application_type.ts
export interface ApplicationType {
    name: string;
    description: string;
    id: number;
    who_can: string[]
}

export interface ApplicationSubType {
    name: string;
    application_type_id: number;
    extra: any;
    id: number;
}

export interface ApplicationTypeResponse extends ApplicationType {
    application_sub_type: ApplicationSubType[];
}